src/utils/data.ts

```diff
@@ -0,0 +1,184 @@
+import { useBenchRunStore } from '../stores/benchRunStore'
+import { useCacheFileStore } from '../stores/cacheFileStore'
+import { useCommitStore } from '../stores/commitStore'
+import { useDataPointStore } from '../stores/dataPointsStore'
+import { useLabelStore } from '../stores/labelStore'
+
+export interface Commit {
+  id: string
+  message: string
+  author: string
+  url: string
+  timestamp: Date
+}
+
+export interface BenchRun {
+  id: string
+  htmlUrl: string
+  runAttempt: number
+  event: string
+  displayTitle: string
+  headCommit: Commit
+  branch: string
+}
+
+export interface BenchDataPoint {
+  label: string
+  score: number
+  benchRun: BenchRun
+}
+
+export function dataPointHash(dp: BenchDataPoint): string {
+  return `${dp.label}-${dp.score}-${dp.benchRun.id}`
+}
+
+async function processSingleFile(content: string): Promise<void> {
+  const commitStore = useCommitStore()
+  const benchRunStore = useBenchRunStore()
+  const dataPointStore = useDataPointStore()
+  const labelStore = useLabelStore()
+
+  const jsObj = JSON.parse(content)
+  const benchRun = jsObj['bench_run']
+  const benchRunId = benchRun['id']
+  const htmlUrl = benchRun['html_url']
+  const runAttempt = benchRun['run_attempt']
+  const event = benchRun['event']
+  const displayTitle = benchRun['display_title']
+  const headCommit = benchRun['head_commit']
+  const commitId = headCommit['id']
+  const commitMsg = headCommit['message']
+  const commitAuthor = headCommit['author']['name']
+  const commitTimestamp = new Date(headCommit['timestamp'])
+  const commitUrl = headCommit['url']
+  const commit: Commit = {
+    id: commitId,
+    message: commitMsg,
+    author: commitAuthor,
+    url: commitUrl,
+    timestamp: commitTimestamp,
+  }
+  if (commitStore.findCommitByID(commitId) === null) {
+    commitStore.addCommit(commit)
+  }
+  const benchRunModel: BenchRun = {
+    id: benchRunId,
+    htmlUrl: htmlUrl,
+    runAttempt: runAttempt,
+    event: event,
+    displayTitle: displayTitle,
+    headCommit: commit,
+    branch: 'develop',
+  }
+  if (benchRunStore.findBenchRunById(benchRunId) === null) {
+    benchRunStore.addBenchRun(benchRunModel)
+  }
+  const scoreDict = jsObj['label_score_dict']
+  for (const [label, score] of Object.entries(scoreDict)) {
+    if (!labelStore.containsLabel(label)) {
+      labelStore.addLabel(label)
+    }
+    const scoreNum = Number.parseFloat(score as string)
+    const dataPoint: BenchDataPoint = {
+      label: label,
+      score: scoreNum,
+      benchRun: benchRunModel,
+    }
+    if (!dataPointStore.containsDatapoint(dataPoint)) {
+      dataPointStore.addDataPoint(dataPoint)
+    }
+  }
+}
+
+/**
+ * Fetches the index file.
+ */
+export async function loadIndex(): Promise<Index> {
+  const indexResp = await fetch(`${FS_URL}/cache/index.json`)
```

Comment:
I don't see any `index.json` file in this PR!

That must mean this code is reading [already existing](https://github.com/enso-org/engine-benchmark-results/commit/7941e22441e3c597163e8dcae0936c4296356d43) `index.json` API in its already existing format.

Keeping the format compatible with currently existing one is good. Thanks.

Reply:
Yes, it is reading the already existing [cache/index.json](https://github.com/enso-org/engine-benchmark-results/blob/7941e22441e3c597163e8dcae0936c4296356d43/cache/index.json). There is nothing special about it - it is a mapping of file names to their timestamps.